src/satctl.c

```diff
@@ -245,10 +273,27 @@ int main(int argc, char **argv)
 	}
 
 	while (csp_zmqhub_idx > 0) {
-		char * zmq_str = csp_zmqhub_addr[--csp_zmqhub_idx];
-		printf("zmq str %s\n", zmq_str);
+		char *zmq_str = csp_zmqhub_addr[--csp_zmqhub_idx];
+		char delimiter[] = ":";
+		int sub_port = CSP_ZMQPROXY_SUBSCRIBE_PORT;
+		int pub_port = CSP_ZMQPROXY_PUBLISH_PORT;
+
+		/* This is how strtok works, each step after first strips the next port number */
```

Comment:
I think people can read the strtok manual if they want to know how it works.
You could document what the intention is however, e.g.: The first invocation of strtok extracts the hostname. The two following invocations extract the port numbers. Note that we specify NULL to continue with the string specified by the first invocation.